File: apps/api/src/domain/identity/value-objects/ConsentType.ts
export enum ConsentTypeValue {
  PERSONAL_DATA = 'personal_data',
  COMMUNICATIONS = 'communications',
  TELEGRAM = 'telegram',
  REVIEW_PUBLICATION = 'review_publication',
}

export class ConsentType {
  private constructor(public readonly value: ConsentTypeValue) {}

  static readonly PERSONAL_DATA = new ConsentType(ConsentTypeValue.PERSONAL_DATA);
  static readonly COMMUNICATIONS = new ConsentType(ConsentTypeValue.COMMUNICATIONS);
  static readonly TELEGRAM = new ConsentType(ConsentTypeValue.TELEGRAM);
  static readonly REVIEW_PUBLICATION = new ConsentType(ConsentTypeValue.REVIEW_PUBLICATION);

  static fromValue(value: string): ConsentType {
    switch (value) {
      case ConsentTypeValue.PERSONAL_DATA:
        return ConsentType.PERSONAL_DATA;
      case ConsentTypeValue.COMMUNICATIONS:
        return ConsentType.COMMUNICATIONS;
      case ConsentTypeValue.TELEGRAM:
        return ConsentType.TELEGRAM;
      case ConsentTypeValue.REVIEW_PUBLICATION:
        return ConsentType.REVIEW_PUBLICATION;
      default:
        throw new Error(`Unknown consent type: ${value}`);
    }
  }

  equals(other: ConsentType): boolean {
    return this.value === other.value;
  }
}

File: backend/domain/identity/value_objects/consent_type.py
"""
ConsentType Value Object.
"""
from domain.shared.value_object import ValueObject


class ConsentType(ValueObject):
    """Value Object для типа согласия."""
    
    def __init__(self, value: str):
        if value not in [
            'personal_data',
            'communications',
            'telegram',
            'review_publication'
        ]:
            raise ValueError(f"Invalid consent type: {value}")
        self._value = value
    
    @property
    def value(self) -> str:
        return self._value
    
    # Предопределенные типы
    PERSONAL_DATA = None
    COMMUNICATIONS = None
    TELEGRAM = None
    REVIEW_PUBLICATION = None


ConsentType.PERSONAL_DATA = ConsentType('personal_data')
ConsentType.COMMUNICATIONS = ConsentType('communications')
ConsentType.TELEGRAM = ConsentType('telegram')
ConsentType.REVIEW_PUBLICATION = ConsentType('review_publication')
